### PR TITLE
Update login flow error handling

### PR DESCRIPTION
**Pull request by suphero**
### Details

This pull request updates error handling on the login flow to improve the identification of errors and their proper display.

### Fixed Issues

$ https://github.com/Expensify/App/issues/16874
PROPOSAL: https://github.com/Expensify/App/issues/16874#issuecomment-1495712532

#### Scenario 1 - Invalid Phone number 

1. If logged in, log out of the app.
2. Enter "1234567890" into the "Phone or email" field.
3. Press the "Continue" button.
4. Ensure the app underlines the "Phone or email" field in red.

Success scenario:

1. Enter the phone number of an existing account.
2. Press the "Continue" button.
3. Ensure the app navigates to the "Password" page.

#### Scenario 2 - Invalid Password

1. If logged in, log out of the app.
2. Enter a valid phone number or email into the "Phone or email" field.
3. Press the "Continue" button.
4. Enter an invalid password into the "Password" field.
5. Press the "Sign In" button.
6. Ensure the app underlines the "Password" field in red.

Success scenario:

1. Enter the valid password into the "Password" field.
2. Press the "Sign In" button.
3. Ensure the app successfully signs in.

#### Scenario 3 - Invalid Two-Factor Code

1. If logged in, log out of the app.
2. Enter the email/phone of an account with TFA enabled into the "Phone or email" field.
4. Press the "Continue" button.
5. Enter the valid password into the "Password" field.
6. Press the "Sign In" button.
7. Enter an invalid two-factor code into the "Two factor code" field.
8. Press the "Sign In" button.
9. Ensure the app underlines only the "Two factor code" field in red.

Success scenario:

1. Enter the valid two-factor code into the "Two factor code" field.
2. Press the "Sign In" button.
3. Ensure the app successfully signs in.

#### Scenario 4 - Invalid Magic code

1. If logged in, log out of the app.
2. Enter the email/phone of an account with Passwordless input enabled ([beta feature](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#working-on-beta-features)) into the "Phone or email" field.
4. Enter an invalid magic code into the "Magic code" field.
5. Press the "Sign In" button.
6. Ensure the app underlines the "Magic code" field in red.

Success scenario:

1. Enter the valid magic code into the "Magic code" field.
2. Press the "Sign In" button.
3. Ensure the app successfully signs in.

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A, this flow requires being online

### QA Steps

Same as Tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

#### Scenario 1 - Invalid Phone number:
web

#### Scenario 2 - Invalid Password
web

#### Scenario 3 - Invalid Two-Factor Code
web

#### Scenario 4 - Invalid Magic code
web

</details>

<details>
<summary>Mobile Web - Chrome</summary>

#### Scenario 1 - Invalid Phone number:
android-chrome

#### Scenario 2 - Invalid Password
android-chrome

#### Scenario 3 - Invalid Two-Factor Code
android-chrome

#### Scenario 4 - Invalid Magic code
android-chrome

</details>

<details>
<summary>Mobile Web - Safari</summary>

#### Scenario 1 - Invalid Phone number:
ios-safari

#### Scenario 2 - Invalid Password
ios-safari

#### Scenario 3 - Invalid Two-Factor Code
ios-safari

#### Scenario 4 - Invalid Magic code
ios-safari

</details>

<details>
<summary>Desktop</summary>

#### Scenario 1 - Invalid Phone number:
desktop

#### Scenario 2 - Invalid Password
desktop

#### Scenario 3 - Invalid Two-Factor Code
desktop

#### Scenario 4 - Invalid Magic code
desktop

</details>

<details>
<summary>iOS</summary>

#### Scenario 1 - Invalid Phone number:
ios

#### Scenario 2 - Invalid Password
ios

#### Scenario 3 - Invalid Two-Factor Code
ios

#### Scenario 4 - Invalid Magic code
ios

</details>

<details>
<summary>Android</summary>

#### Scenario 1 - Invalid Phone number:
android

#### Scenario 2 - Invalid Password
android

#### Scenario 3 - Invalid Two-Factor Code
android

#### Scenario 4 - Invalid Magic code
android

</details>

### CODE REVIEW

**Comment by suphero** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by suphero** [comment]
recheck

**Comment by cristipaval** [comment]
@suphero next time please link the issue and the comment with the proposal when you create the PR, in order to get the appropriate reviewers assigned.

**Comment by suphero** [comment]
@cristipaval, I've already linked the issue and the comment with the proposal. Is there any other step that I missed?

Screenshot 2023-04-10 at 16 13 01

**Comment by cristipaval** [comment]
> @cristipaval, I've already linked the issue and the comment with the proposal. Is there any other step that I missed?
> 
> Screenshot 2023-04-10 at 16 13 01

Did you link them at the PR creation or afterwards?

**Comment by cristipaval** [comment]
Hmm... @mountiny I guess Melvin failed this time, or?

**Comment by suphero** [comment]
Oh I see, I'll be careful next time

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]
The password page is not handled here, check this https://github.com/Expensify/App/issues/16874#issuecomment-1496450488

**Comment by Santhosh-Sellavel** [comment]
@flodnv We should handle this and validate the code form too right?

https://user-images.githubusercontent.com/85645967/230988701-dfd8a553-7da0-4e5d-9b26-c8c86a7114af.mov

cc: @suphero

**Comment by flodnv** [comment]
Yes this makes sense to me 👍

**Comment by Santhosh-Sellavel** [comment]
Cool, @suphero Can you make the change and update PR details accordingly thanks!

**Comment by suphero** [comment]
> Cool, @suphero Can you make the change and update PR details accordingly thanks!

I've made the change and updated the PR.

**Comment by mountiny** [comment]
image

I think it was not initially linked as a full link

@suphero for next prs, make sure its a full link after the dollar sign `$ https://github.com/Expensify/App/pull/17168`

Thank you 🙇 Great to see a first time contributor!! 🎉

**Comment by suphero** [comment]
> image
> 
> I think it was not initially linked as a full link
> 
> @suphero for next prs, make sure its a full link after the dollar sign `$ https://github.com/Expensify/App/pull/17168`
> 
> Thank you 🙇 Great to see a first time contributor!! 🎉

Thanks @mountiny, I'll keep it in mind.

**Comment by Santhosh-Sellavel** [comment]
@suphero password page is issued here

**Comment by suphero** [comment]
> @suphero password page is missed here

You mean in description or changed files? I think both requirements are met.

**Comment by Santhosh-Sellavel** [comment]
Seems we've run into conflicts as well

**Comment by suphero** [comment]
> Seems we've run into conflicts as well

I've already handled the magic code red line in the PR. I'll fix the conflict in a few hours.

**Comment by Santhosh-Sellavel** [comment]
Check this one https://github.com/Expensify/App/pull/17168#pullrequestreview-1378061426

**Comment by suphero** [comment]
I've updated the PR. I have a question about the last change.

When the two-factor is enabled, the app requires the two-factor code after valid password input (Scenario 3). If an invalid code is entered, the API returns an error. Since the error may also be related to an internal server error, it's a better practice to underline both of the fields ("Password" and the "Two factor code") with red color. If the user code does not pass the regex, only the "Two factor code" field is underlined with red color. Is it ok to proceed this way?

API Error:
android

Regex Error: 
android

**Comment by flodnv** [comment]
Hmmm, I think it's better in both scenarios to only highlight in red the 2FA code field, because the user has already passed/completed the password field.

**Comment by suphero** [comment]
Ok, I’ll update the PR

**Comment by suphero** [comment]
> Hmmm, I think it's better in both scenarios to only highlight in red the 2FA code field, because the user has already passed/completed the password field.

I've updated the PR

File: src/pages/signin/LoginForm.js [hunk 1]
```diff
@@ -166,6 +166,7 @@ class LoginForm extends React.Component {
     render() {
         const formErrorText = this.state.formError ? this.props.translate(this.state.formError) : '';
         const serverErrorText = ErrorUtils.getLatestErrorMessage(this.props.account);
+        const hasError = !_.isEmpty(formErrorText) || !_.isEmpty(serverErrorText);
```
Comment (Santhosh-Sellavel) [inline src/pages/signin/LoginForm.js:169]: `!_.isEmpty(formErrorText)`

This here is kind of redundant as this already handled if `errorText` is there errorHighlight will be shown

File: src/pages/signin/PasswordForm.js [hunk 2]
```diff
@@ -168,6 +168,14 @@ class PasswordForm extends React.Component {
     }
 
     render() {
+        const formHasPasswordError = !_.isEmpty(this.state.formError.password);
+        const formHasTwoFactorError = !_.isEmpty(this.state.formError.twoFactorAuthCode);
```
Comment (Santhosh-Sellavel) [inline src/pages/signin/PasswordForm.js:172]: Same here

File: src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js [hunk 3]
```diff
@@ -189,6 +189,8 @@ class BaseValidateCodeForm extends React.Component {
     }
 
     render() {
+        const formHasMagicCodeError = !_.isEmpty(this.state.formError.validateCode);
```
Comment (Santhosh-Sellavel) [inline src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js:192]: Same here

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

File: src/pages/signin/LoginForm.js (hunk 1 above)
Comment (Santhosh-Sellavel) [reply src/pages/signin/LoginForm.js:169]: Your thoughts @flodnv?

Comment (suphero) [reply src/pages/signin/LoginForm.js]: You're right, fixed that

File: src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js [hunk 4]
```diff
@@ -189,6 +189,7 @@ class BaseValidateCodeForm extends React.Component {
     }
 
     render() {
+        const hasError = Boolean(this.props.account) && !_.isEmpty(this.props.account.errors);
         return (
             <>
```
Comment (Santhosh-Sellavel) [inline src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js:194]: Two Factor Authentication Missed here

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

File: src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js (hunk 4 above)
Comment (suphero) [reply src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js:194]: I couldn't find a way to pass the magic code. I'm changing the following line to return true, but it doesn't allow me to pass the magic code for the 2FA validation.

https://github.com/Expensify/App/blob/af70dc29746608efeb1c4c95c0a0dca731bfe122/src/libs/Permissions.js#L94

Comment (suphero) [reply src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js:194]: I committed the solution method that I can't guarantee.

Comment (Santhosh-Sellavel) [reply src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js:194]: That's behind beta, for I'll test and let you know!

Comment (suphero) [reply src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js:194]: Thanks, I'm waiting

**Comment by Santhosh-Sellavel** [comment]
### Screenshots/Videos
<details>
<summary>Web & Desktop </summary>

https://user-images.githubusercontent.com/85645967/232165374-9885512e-804b-4999-9516-8c7a4e4db052.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/85645967/232168634-df397aab-f376-4c2f-abc5-0f9a0b15880d.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/85645967/232168602-4a0d4588-b5a5-445f-bad9-f5ba6d112b99.mp4

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/85645967/232168036-ad7da8be-15ed-478c-ac9e-b0b62c4a6c2b.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/85645967/232168075-532e7a1c-af08-4e8a-a736-21513db55825.mp4

</details>

**Comment by Santhosh-Sellavel** [comment]
Tests well mostly, I just run into a weird case (Kinda rare scenario)

1. User Entered a valid email
2. User Entered the valid password
3. Now TFA is required
4. At this we allow users to modify passwords also, when modified to a invalid password and entering a valid TFA also highlights TFA with Red Line but messages says incorrect password

https://user-images.githubusercontent.com/85645967/232168712-8b07928b-05b8-464d-8eb0-e4c5473ea7ee.mp4


@flodnv What should we do we here, should we ignore this or handle it?

Why do we allow to modify the password here? 

Can we just disable modifying passwords or just show TFA after the password is successful the first time? i.e as we do in the Magic code Flow?

cc: @suphero

**Comment by Santhosh-Sellavel** [comment]
@suphero 

These steps could be improved, I've just pointed out a few make sure it's fixed in all places thanks

> Enter valid input  or Enter any input

Can be more specific here, i.e Enter a valid phone number or email

> Enter any number that can log in.

Enter the phone number of the existing account

> Enter any input into the "Phone or email" field that supports two-factor authentication.

Enter the email/phone of the account with TFA enabled into the "Phone or email

**Comment by suphero** [comment]
I have improved the PR scenario steps for any unclear content. I'm waiting for the decision on the [rare scenario](https://github.com/Expensify/App/pull/17168#issuecomment-1509382479).

**Comment by flodnv** [comment]
Let's not worry about the rare case as we're about to remove passwords from everywhere in the next few weeks.

**Comment by Santhosh-Sellavel** [comment]
@flodnv Can you run workflows

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Comment by Santhosh-Sellavel** [comment]
@suphero 

Look Have a look at the to this [thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1660778311856709) sign commits

**Review by Santhosh-Sellavel — COMMENTED** [review]
LGTM tests well, to waiting to sign commits

**Comment by suphero** [comment]
@flodnv @Santhosh-Sellavel Can you please run workflows?

**Review by Santhosh-Sellavel — DISMISSED** [review]
LGTM, all you!

@flodnv Can you run the workflow again?

**Review by flodnv — COMMENTED** [review]
Just one minor comment, looks good 👍

**Comment by suphero** [comment]
Fixed [the minor comment](https://github.com/Expensify/App/pull/17168#pullrequestreview-1388967641), can you please run the workflow again?

@flodnv @Santhosh-Sellavel

File: src/pages/signin/PasswordForm.js [hunk 5]
```diff
@@ -168,6 +168,12 @@ class PasswordForm extends React.Component {
     }
 
     render() {
+        const isTwoFactorAuthRequired = Boolean(this.props.account.requiresTwoFactorAuth);
+        const serverHasError = Boolean(this.props.account) && !_.isEmpty(this.props.account.errors);
```
Comment (Santhosh-Sellavel) [inline src/pages/signin/PasswordForm.js:172]:
```suggestion
        const hasServerError = Boolean(this.props.account) && !_.isEmpty(this.props.account.errors);
```

Comment (Santhosh-Sellavel) [reply src/pages/signin/PasswordForm.js:172]: Could better name here

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]
Minor, thing!

**Comment by suphero** [comment]
> Minor, thing!

Fixed that too.

**Review by flodnv — APPROVED** [review]

**Comment by suphero** [comment]
Is there any issue with the workflow jobs? It seems that workflows were stuck at 99.7% for 30 minutes.

Received 1448274937 of 1452469241 (99.7%)

https://github.com/Expensify/App/actions/runs/4726168929/jobs/8385976224
https://github.com/Expensify/App/actions/runs/4726168929/jobs/8385975889

**Comment by suphero** [comment]
> Is there any issue with the workflow jobs? It seems that workflows were stuck at 99.7% for 30 minutes.
> 
> Received 1448274937 of 1452469241 (99.7%)
> 
> https://github.com/Expensify/App/actions/runs/4726168929/jobs/8385976224 https://github.com/Expensify/App/actions/runs/4726168929/jobs/8385975889

The workflows passed after 1 hour with the following message.

```
Warning: Failed to restore: Aborting cache download as the download time exceeded the timeout.
npm cache is not found
```

**Comment by Santhosh-Sellavel** [comment]
@flodnv if workflows pass feel free merge this one!

**Pull request merged**